specifyweb/specify/tree_extras.py

```diff
@@ -69,10 +69,40 @@ def save():
         try:
             model.objects.get(id=self.id, parent__rankid__lt=F('rankid'))
         except model.DoesNotExist:
-            raise BusinessRuleException("Tree node's parent has rank greater than itself.")
+            raise TreeBusinessRuleException(
+                "Tree node's parent has rank greater than itself",
+                {"tree" : self.__class__.__name__,
+                 "type" : "TREE_RANK_INVARIANT_PARENT",
```

Comment:
Why do you need both type and localizationKey? Can we just have localizationKey?

Reply:
If you need it just so that the error in the UI is in upper case, then just add a function called `camelToUpper` in here and call it when printing the error code: https://github.com/specify/specify7/blob/c8a6773f746491141a02f0af216a008c12c63b90/specifyweb/frontend/js_src/lib/utils/utils.ts#L29-L42

Reply:
Fixed in 44f6b8e